packages/extension-api/src/extension-api.d.ts

```diff
@@ -2295,4 +2295,16 @@ declare module '@podman-desktop/api' {
     // eslint-disable-next-line @typescript-eslint/no-explicit-any
     export function setValue(key: string, value: any, scope?: 'onboarding'): void;
   }
+
+  export interface CliToolOptions {
+    id: string;
+    name: string;
+    displayName: string;
+    description: string;
+    images: ProviderImages;
+  }
+
+  export namespace cli {
+    export function createCliTool(options: CliToolOptions): Disposable;
```

Comment:
I think we should go with the `registerCliToolProvider` approach and providing a `CliTool`
then we can return a Disposable

if we have a `create***()` method, an object should be returned and not a Disposable (return CliTool object)

but it looks more to be a `provider` thing to me

Reply:
I am going to stick to create method returning CliTool, the same pattern used for Providers.

Reply:
@dgolovin providers use a registration mechanism, not a create method

Reply:
for example registerProvider vs createStatusbarItem

when we provide something, it's a provider (then a register thing), when we ask the engine to create object for us like a StatusBar, a Messagebox, etc it's a create

but here it's a registerCliToolProvider where you provide the (cliToolProvider object)

Reply:
I'm a bit lost on 
> I am going to stick to create method returning CliTool, the same pattern used for Providers.

but on https://github.com/containers/podman-desktop/pull/4197#issuecomment-1752429026
![image](https://github.com/containers/podman-desktop/assets/436777/0b563734-7672-4e99-b269-990f39ff8cbd)

it's using register approach so I would say you'll stick to a register method returning a Disposable object as shown in the image ?

Reply:
I meant to use approach similar to provider module here:
https://github.com/containers/podman-desktop/blob/efbe62aef78467ac79644839d69d7a3a9a89c327/packages/extension-api/src/extension-api.d.ts#L426

Right now stripped diagram looks like this:
<img src="https://github.com/containers/podman-desktop/assets/620330/8c54eaf1-f218-41cf-a08b-b57c1dc83bbe" width="400px" />

Currently implemented on UI side:
1. Simple UI
2. UI update on 'extension-loaded' event
3. UI update on 'cli-tool-create' or 'cli-tool-remove'

On main side:
1. Simple registry above
2. Notification on create and dispose of CliTool
3. Generating of the Info objects for UI
4. Cli Tool declaration in package.json through extension

You can ran it from branch and you will see docker-compose in CLI Tools preferences
If you install/uninstall `ghcr.io/dgolovin/podman-desktop-kubectl-ext:latest` you will see kubectl appears and disappears in UI. Compose is using cli module to define cli tool form activation method. kubctl does it in package.json extension manifest.

https://github.com/containers/podman-desktop/assets/620330/0840e5d4-62ac-48b2-bae4-e0ec7fde31b3